nbclient/client.py

```diff
@@ -223,6 +223,35 @@ class NotebookClient(LoggingConfigurable):
 
     kernel_manager_class = Type(config=True, help='The kernel manager class to use.')
 
+    on_kernel_create = Any(
```

Comment:
This might be deceptive to users, because it's only run when nbclient makes a kernelmanager and not when a kernel is created. The kernel creation aspect is fairly abstracted away from nbclient so I would instead make a notebook_start hook after the kernel setup is completed if you're going for pre-cell execution hooks.